src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -433,10 +431,22 @@ protected override int MakeSlot(BoundExpression node)
                                         //   struct B { }
                                         //   if (a?.B != null) _ = ((A)a).B.Value; // no warning
                                         int containingSlot = MakeSlot(operand);
-                                        return containingSlot < 0 ? -1 : GetNullableOfTValueSlot(operandType, containingSlot);
+                                        return containingSlot < 0 ? -1 : GetNullableOfTValueSlot(operandType, containingSlot, out _);
+                                    }
+                                    else if (AreNullableAndUnderlyingTypes(convertedType, operandType, out _))
+                                    {
+                                        // Explicit conversion of T to Nullable<T> is equivalent to new Nullable<T>(t).
```

Comment:
>// Explicit conversion of T to Nullable<T> is equivalent to new Nullable<T>(t). [](start = 40, length = 79)

Shouldn't conversion from ```T``` to ```Nullable<T>``` be classified as implicit?

Reply:
`(S?)new S()` is classified as an `ExplicitNullable` conversion.

---
In reply to: [250776752](https://github.com/dotnet/roslyn/pull/32729#discussion_r250776752) [](ancestors = 250776752)

Reply:
This feels wrong, classification should reflect the nature of conversion , not the syntactic form used. Please open an issue for this.

---
In reply to: [250778640](https://github.com/dotnet/roslyn/pull/32729#discussion_r250778640) [](ancestors = 250778640,250776752)